assets/src/application/ui/display/confirm/useConfirm.tsx

```diff
@@ -1,12 +1,15 @@
-import React, { useCallback, useMemo } from 'react';
-import { Button, useDisclosure } from '@chakra-ui/core';
+import React, { useCallback } from 'react';
+import { useDisclosure } from '@chakra-ui/core';
```

Comment:
I also have used  `useDisclosure` in `Confirm` comp in the app layer, but not sure what's the best approach, if we want to keep external deps at this layer or not, just thinking outloud :thinking:

Reply:
If it was a separate package that could be installed on its own then that would be fine, but as is, if we want/need to change UI library in the future, then this will need refactoring. It's ok to directly import low-level utility packages like Ramda and date-fns but for components we should use adapters. This kinda falls in between but because it is part of a UI library, should be decoupled imho.

potential options:

- create an adapter for `useDisclosure`

- build our own custom hook(s) for managing common state patterns

- use another library for non-UI utilities like: https://hooks.umijs.org/hooks/state/use-toggle

Reply:
+1 for adapter, the easiest and with less friction option ATM

Reply:
I agree with creating an adapter